aptos-move/block-executor/src/scheduler.rs

```diff
@@ -570,6 +571,52 @@ impl Scheduler {
         SchedulerTask::NoTask
     }
 
+    pub fn finish_execution_during_commit(&self, txn_idx: TxnIndex, revalidate_suffix: bool) {
+        // Note: It is preferable to hold the validation lock throughout the finish_execution,
+        // in particular before updating execution status. The point was that we don't want
+        // any validation to come before the validation status is correspondingly updated.
+        // It may be possible to make work more granular, but shouldn't make performance
+        // difference and like this correctness argument is much easier to see, in fact also
+        // the reason why we grab write lock directly, and never release it during the whole function.
+        // So even validation status readers have to wait if they somehow end up at the same index.
+        let _validation_status = self.txn_status[txn_idx as usize].1.write();
```

Comment:
we shouldn't need this lock, as we wouldn't be updating validation status for this index (we the committing thread control everything here).